ts/index.ts

```diff
@@ -170,6 +184,18 @@ class Gossipsub extends BasicPubsub {
     // Add to peer scoring
     this.score.addPeer(peerId.toB58String())
 
+    // track the connection direction
+    let outbound = false
+    for (const c of this._connectionManager.getAll(peerId)) {
+      if (c.stat.direction === 'outbound') {
+        if (Array.from(c.registry.values()).some(rvalue => protocols.includes(rvalue.protocol))) {
```

Comment:
This isn't great, not sure what the best way to do this is.
Trying to ensure that gossipsub is one of the protocols of one of the streams on the connection (considering that there may be multiple connections for a peer, not all containing a gossipsub stream)

Reply:
humm, yeah!
Maybe the connection could provide an utility API to check if it has open streams using a given protocol.
I will add to my TODO list to consider and implement it. Meanwhile, we can go this way